Symfony/CS/Tests/Fixer/Contrib/Php4ConstructorFixerTest.php

```diff
@@ -0,0 +1,521 @@
+ */
+
+namespace Symfony\CS\Tests\Fixer\Contrib;
+
+use Symfony\CS\Tests\Fixer\AbstractFixerTestBase;
+
+/**
+ * @author Matteo Beccati <matteo@beccati.com>
+ */
+class Php4ConstructorFixerTest extends AbstractFixerTestBase
+{
+    public function testNamespaces()
+    {
+        $input = <<<'EOF'
+<?php
+
+namespace Baz;
+
+class Foo
+{
+    public function __construct($bar)
+    {
+        var_dump(1);
+    }
+
+    public function Foo($bar)
+    {
+        var_dump(2);
+    }
+}
+EOF;
+
+        $this->makeTest($input);
+    }
+
+    public function testNamespaces2()
+    {
+        $input = <<<'EOF'
+<?php
+
+namespace Baz
+{
+    class Foo
+    {
+        public function Foo($bar)
+        {
+            var_dump(1);
+        }
+
+        public function Foo($bar)
+        {
+            var_dump(2);
+        }
+    }
+}
+
+class Foo
+{
+    public function __construct($bar)
+    {
+        var_dump(2);
+    }
+}
+EOF;
+
+        $expect = <<<'EOF'
+<?php
+
+namespace Baz {
+
+    class Foo
+    {
+        public function __construct($bar)
+        {
+            var_dump(1);
+        }
+
+        public function Foo($bar)
+        {
+            var_dump(2);
+        }
+    }
+}
+
+class Foo
+{
+    public function __construct($bar)
+    {
+        var_dump(2);
+    }
+}
+EOF;
+
+        $this->makeTest($input);
+    }
+
+    public function testPhp5Only()
+    {
+        $input = <<<'EOF'
+<?php
+
+class Foo
+{
+    function __construct($bar)
+    {
+        var_dump(1);
+    }
+
+    function bar()
+    {
+        var_dump(3);
+    }
+}
+EOF;
+
+        $this->makeTest($input);
+    }
+
+    public function testPhp4Only()
+    {
+        $input = <<<'EOF'
+<?php
+
+class Foo
+{
+    /**
+     * Constructor
+     */
+    function Foo($bar)
+    {
+        var_dump(1);
+    }
+
+    function bar()
+    {
+        var_dump(3);
+    }
+}
+EOF;
+
+        $expect = <<<'EOF'
+<?php
+
+class Foo
+{
+    /**
+     * Constructor
+     */
+    function __construct($bar)
+    {
+        var_dump(1);
+    }
+
+    function bar()
+    {
+        var_dump(3);
+    }
+}
+EOF;
+
+        $this->makeTest($expect, $input);
+    }
+
+    public function testBothRight()
+    {
+        $input = <<<'EOF'
+<?php
+
+class Foo
+{
+    /**
+     * Constructor
+     */
+    function __construct($bar)
+    {
+        var_dump(1);
+    }
+
+    /**
+     * PHP-4 Constructor
+     */
+    function Foo($bar)
+    {
+        $this->__construct($bar);
+    }
+
+    function bar()
+    {
+        var_dump(3);
+    }
+}
+EOF;
+
+        $expect = <<<'EOF'
+<?php
+
+class Foo
+{
+    /**
+     * Constructor
+     */
+    function __construct($bar)
+    {
+        var_dump(1);
+    }
+
+    function bar()
+    {
+        var_dump(3);
+    }
+}
+EOF;
+
+        $this->makeTest($expect, $input);
+    }
+
+    public function testBothWrong()
+    {
+        $input = <<<'EOF'
+<?php
+
+class Foo
+{
+    /**
+     * PHP-5 Constructor
+     */
+    function __construct($bar)
```

Comment:
And that behavior is wrong.

take a look:
```php
<?php

class Foo
{
    function __construct()
    {
        echo "constructor";
    }

    function Foo()
    {
        echo "non-related";
    }
}
```

Let sey that I use PHP5 and never seen PHP4.
I create a class with constructor.
`$f = new Foo();` will print `constructor`. Great.

Then I add `Foo` method (not a constructor, I am writing in PHP5), and then:
`$f->Foo();` will print `non-related`. That is also great!

And your fixer will remove my method!